examples/rust/publish_subscribe_cross_language/README.md

```diff
@@ -0,0 +1,132 @@
+cmake --build target/ffi/build
+```
+
+To observe this dynamic communication in action, open two separate terminals and
+execute the following commands:
+
+### Terminal 1
+
+Run the C++ subscriber application from the
+[C++ cross-language publish-subcribe example](../../cxx/publish_subscribe_cross_language):
+
+```sh
+./target/ffi/build/examples/cxx/publish_subscribe_cross_language/example_cxx_publish_subscribe_cross_language_subscriber
+```
+
+### Terminal 2
+
+Run the Rust publisher application:
+
+```sh
+cargo run --example publish_subscribe_cross_language_publisher
+```
+
+Feel free to also run the Rust subscriber and the C++ publisher applications
+simultaneously to explore how iceoryx2 handles cross-language
+publisher-subscriber communication efficiently.
+
+### Terminal 3
+
+Run the Rust subscriber application:
+
+```sh
+cargo run --example publish_subscribe_cross_language_subscriber
+```
+
+### Terminal 4
+
+Run the C++ publisher application from the
+[C++ cross-language publish-subscribe example](../../cxx/publish_subscribe_cross_language):
+
+```sh
+./target/ffi/build/examples/cxx/publish_subscribe_cross_language/example_cxx_publish_subscribe_cross_language_publisher
+```
+
+You can also communicate with the C publisher and subscriber applications from
+the
+[C cross-language publish-subscribe example](../../c/publish_subscribe_cross_language).
+
+You may hit the maximum supported number of ports when too many publisher or
+subscriber processes are running. Check the [iceoryx2 config](../../../config)
+to set the limits globally or refer to the
+[API of the Service builder](https://docs.rs/iceoryx2/latest/iceoryx2/service/index.html)
+to set them for a single service.
+
+## How to enable publish-subscribe communication between Rust, C++ and C applications
+
+To communicate with each other, publisher and subscriber applications must share
+the same service configuration, including the payload and the user header type
+name.
+
+For C++ and Rust applications, the internally derived type names
+usually depend on the used programming language. To allow cross-language
+communication involving C++ applications, iceoryx2 provides the possibility to
+customize the payload and the user header type name by setting `TYPE_NAME` in
+the sent C++ data struct and user header, e.g.
+
+```cxx
+struct TransmissionData {
+    static constexpr const char* TYPE_NAME = "examples_common::transmission_data::TransmissionData";
+    std::int32_t x;
+    std::int32_t y;
+    double funky;
+};
+
+struct CustomHeader {
+    static constexpr const char* TYPE_NAME = "examples_common::custom_header::CustomHeader";
+    int32_t version;
+    uint64_t timestamp;
+};
+```
+
+For C applications, these type names must be set with
+`iox2_service_builder_pub_sub_set_payload_type_details` and
+`iox2_service_builder_pub_sub_set_user_header_type_details` before creating the
+service.
+
+_Note:_ The type name can't currently be set for Rust applications. You can
```

Comment:
Please use the github markdown extension for the not -> https://github.com/orgs/community/discussions/16925